### PR TITLE
refactor(core): refactor the sending and support `spoiler`

### PR DESCRIPTION
**Pull request by Lipraty**
> [!WARNING]
> waiting for updates from upstream

On the Telegram platform, the combination of `<spl><image/></spl>` is not supported, and this method is only applicable to text. However, the image API does support the `has_spoiler` option, so we can push for the standardization of this method in Satori Element: `<image spoiler/>`.

Update: Koishi RFC: https://github.com/koishijs/koishi/issues/1258

### CODE REVIEW

**Comment by MaikoTan** [comment]
Is there a link for the upstream updates, an issue or a pending PR? Please include it into the description so we could track it from here.

File: packages/core/src/index.tsx [hunk 1]
```diff
@@ -153,7 +160,12 @@ export const Config = Schema.intersect([
       Schema.const(3).description('发送全部信息'),
     ]).description('输出方式。').default(1),
     asset: Schema.boolean().default(false).description('优先使用 [assets服务](https://assets.koishi.chat/) 转存图片。'),
-    base64: Schema.boolean().default(false).description('使用 base64 发送图片。')
+    base64: Schema.boolean().default(false).description('使用 base64 发送图片。'),
+    spoiler: Schema.union([
+      Schema.const(0).description('禁用'),
+      Schema.const(1).description('所有图片'),
+      Schema.const(2).description('仅色图 (NSFW)'),
```
Comment (MaikoTan) [inline packages/core/src/index.tsx:167]: `僅 NSFW`

Comment (Lipraty) [reply packages/core/src/index.tsx:167]: 好

**Comment by MaikoTan** [comment]
TODO:

1. Add a comment at the line 267 that you're checking the current platform, explaining that why we should do this.
1. Mark this option as an `experimental` feature.
1. Add a comprehensive document for the option correspondingly, mostly why and when to use it.

**Comment by Lipraty** [comment]
> 3\. Add a comprehensive document for the option correspondingly, mostly why and when to use it.

The document appears to be lagging behind many configurations. It is recommended that we open another pull request to address this.

**Review by MaikoTan — APPROVED** [review]
LGTM

**Pull request merged**